deployment/src/main/java/org/wso2/testgrid/deployment/deployers/KubernetesDeployer.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.testgrid.deployment.deployers;
+
+import com.google.gson.Gson;
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.client5.http.fluent.Response;
+import org.apache.http.HttpHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.*;
+import org.wso2.testgrid.common.config.ConfigurationContext;
+import org.wso2.testgrid.common.config.DeploymentConfig;
+import org.wso2.testgrid.common.config.Script;
+import org.wso2.testgrid.common.exception.CommandExecutionException;
+import org.wso2.testgrid.common.exception.TestGridDeployerException;
+import org.wso2.testgrid.common.util.DataBucketsHelper;
+import org.wso2.testgrid.common.util.StringUtil;
+import org.wso2.testgrid.deployment.DeploymentUtil;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class performs Kubernetes related deployment tasks.
+ *
+ * @since 1.0.0
+ */
+public class KubernetesDeployer implements Deployer {
+
+    private static final Logger logger = LoggerFactory.getLogger(KubernetesDeployer.class);
+    private static final String DEPLOYER_NAME = "Kubernetes";
+
+    @Override
+    public String getDeployerName() {
+        return DEPLOYER_NAME;
+    }
+
+    /**
+     * This class is used to invoke the script to deploy the deployments in Kubernetes Engine
+     *
+     * @param testPlan current testPlan configurations
+     * @param infrastructureProvisionResult infrastructure provisioning output
+     * @return
+     * @throws TestGridDeployerException
+     */
+    @Override
+    public DeploymentCreationResult deploy(TestPlan testPlan,
+                                           InfrastructureProvisionResult infrastructureProvisionResult)
+            throws TestGridDeployerException {
```

Comment:
below logic does not contain a logic specific to kubernetes. It seems more like a ShellDeployer. Please evaluate the possibility of removing this.

Reply:
Yeah it is, but in this step, the specific script kubernetes_deploy.sh has to be run when the type is kubernetes. So this deploy function is used.

Reply:
I see. To which usecase does this applies to? We only have two usecases at deploy phase, right? shell script (SHELL type)+kubernetes yamls (KUBERNETES type), or just kubernetes yamls (KUBERNETES type). (the script type is mentioned inside brackets).

Out of these two usecase, which one are you talking about here?